public/app/features/explore/state/actions.ts

```diff
@@ -543,7 +545,7 @@ export function refreshExplore(exploreId: ExploreId): ThunkResult<void> {
     // need to refresh datasource
```

Comment:
Maybe we need to check if the `mode` has changed when browsing back/forward in browser i.e. add check for mode and then dispatch correct action accordingly.